pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -418,6 +418,10 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				return err
 			}
 
+			if sdsTokenPath, found := con.modelNode.Metadata["SDS_K8S_TOKEN_PATH"]; found {
```

Comment:
- the metadata key goes to model/context.go, and needs to have documentation.
- I would call it SDS_TOKEN_PATH - I suspect any platform that is capable of creating an equivalent JWT token would work, it's not specific to k8s. 
- the normal pattern is to just use the con.modelNode - it is part of the parameters that plugin is passing. 
Using a static var is not ideal or consistent with how the rest of the code is written.

Reply:
And I'm pretty sure it'll only work if you have a single value - which brings the point that you should add a test, with at least 2 nodes having different configuration ( since the current code seems to have this problem by using the var )